actors/builtin/miner/miner_actor.go

```diff
@@ -547,7 +547,11 @@ func (a Actor) PreCommitSector(rt Runtime, params *PreCommitSectorParams) *abi.E
 	newlyVested := big.Zero()
 	feeToBurn := abi.NewTokenAmount(0)
 	rt.StateTransaction(&st, func() {
-		newlyVested, err = st.UnlockVestedFunds(store, rt.CurrEpoch())
+		// Stop vesting funds as of version 7. Its computationally expensive and unlikely to release any funds.
+		if rt.NetworkVersion() < network.Version7 {
+			newlyVested, err = st.UnlockVestedFunds(store, rt.CurrEpoch())
+		}
+
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to vest funds")
```

Comment:
nit: I think it is fine in practice to keep this out here because err will always be nil but this should go within the if statement.

Reply:
good catch